src/main/java/org/opensearch/sdk/ExtensionsRunner.java

```diff
@@ -132,7 +132,10 @@ protected ExtensionsRunner(Extension extension) throws IOException {
         // save custom transport actions
         this.transportActions = new TransportActions(extension.getActions());
         // initialize the transport service
-        nettyTransport.initializeExtensionTransportService(this.getSettings(), this);
+        ThreadPool threadPool = new ThreadPool(this.getSettings());
+        nettyTransport.initializeExtensionTransportService(this.getSettings(), threadPool, this);
+        // create components
+        extension.createComponents(new SDKClient(), null, threadPool);
```

Comment:
Can we instantiate the client in this class as an instance variable and just pass a reference to the object here?  With this approach if the ExtensionsRunner wanted access to the client it'd not be able to.